src/Aspire.Cli/Templating/DotNetTemplateFactory.cs

```diff
@@ -0,0 +1,329 @@
+{
+    public IEnumerable<ITemplate> GetTemplates()
+    {
+        yield return new CallbackTemplate(
+            "aspire-starter",
+            "Aspire Starter App",
+            projectName => $"./{projectName}",
+            ApplyExtraAspireStarterOptions,
+            (template, parseResult, ct) => ApplyTemplateAsync(template, parseResult, PromptForExtraAspireStarterOptionsAsync, ct)
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire",
+            "Aspire Empty App",
+            projectName => $"./{projectName}",
+            _ => { },
+            ApplyTemplateWithNoExtraArgsAsync
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire-apphost",
+            "Aspire App Host",
+            projectName => $"./{projectName}",
+            _ => { },
+            ApplyTemplateWithNoExtraArgsAsync
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire-servicedefaults",
+            "Aspire Service Defaults",
+            projectName => $"./{projectName}",
+            _ => { },
+            ApplyTemplateWithNoExtraArgsAsync
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire-mstest",
+            "Aspire Test Project (MSTest)",
+            projectName => $"./{projectName}",
+            _ => { },
+            ApplyTemplateWithNoExtraArgsAsync
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire-nunit",
+            "Aspire Test Project (NUnit)",
+            projectName => $"./{projectName}",
+            _ => { },
+            ApplyTemplateWithNoExtraArgsAsync
+            );
+            
+        yield return new CallbackTemplate(
+            "aspire-xunit",
+            "Aspire Test Project (xUnit)",
+            projectName => $"./{projectName}",
+            _ => { },
+            (template, parseResult, ct) => ApplyTemplateAsync(template, parseResult, PromptForExtraAspireXUnitOptionsAsync, ct)
+            );
+    }
+
+    private async Task<string[]> PromptForExtraAspireStarterOptionsAsync(ParseResult result, CancellationToken cancellationToken)
+    {
+        var extraArgs = new List<string>();
+
+        await PromptForRedisCacheOptionAsync(result, extraArgs, cancellationToken);
+        await PromptForTestFrameworkOptionsAsync(result, extraArgs, cancellationToken);
+
+        return extraArgs.ToArray();
+    }
+
+    private async Task<string[]> PromptForExtraAspireXUnitOptionsAsync(ParseResult result, CancellationToken cancellationToken)
+    {
+        var extraArgs = new List<string>();
+
+        await PromptForXUnitVersionOptionsAsync(result, extraArgs, cancellationToken);
+
+        return extraArgs.ToArray();
+    }
+
+    private async Task PromptForRedisCacheOptionAsync(ParseResult result, List<string> extraArgs, CancellationToken cancellationToken)
```

Comment:
oh wow, so this is all hard-coded? Hmm, did we consider automating keeping this up to date via some code-gen or something? @davidfowl @mitchdenny